internal/olm/client/client.go

```diff
@@ -222,7 +224,74 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 		}
 	}
 
-	return wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	err := wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	if err != nil && errors.Is(err, context.DeadlineExceeded) {
+		if depCheckErr := c.printDeploymentErrors(ctx, key, csv); depCheckErr != nil {
+			return fmt.Errorf("error printing operator resource errors: %v %v", err, depCheckErr)
+		}
+	}
+	return err
+}
+
+// TODO(btenneti) Refactor function to collect errors into customized error and return.
+// printDeploymentErrors function loops through deployment specs of a given CSV, and prints reason
+// in case of failures, based on deployment condition.
+func (c Client) printDeploymentErrors(ctx context.Context, key types.NamespacedName, csv olmapiv1alpha1.ClusterServiceVersion) error {
+	if key.Namespace == "" {
+		return fmt.Errorf("no namespace provided to get deployment failures")
+	}
+	dep := &appsv1.Deployment{}
+	for _, ds := range csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+		depKey := types.NamespacedName{
+			Namespace: key.Namespace,
+			Name:      ds.Name,
+		}
+		depSelectors := ds.Spec.Selector
+		if err := c.KubeClient.Get(ctx, depKey, dep); err != nil {
+			log.Printf("error getting operator deployment %q: %v", ds.Name, err)
+			continue
+		}
+		for _, s := range dep.Status.Conditions {
+			if s.Type == appsv1.DeploymentAvailable && s.Status == corev1.ConditionFalse {
+				log.Printf("operator deployment %q not available: %s", ds.Name, s.Reason)
+				if err := c.printPodErrors(ctx, depSelectors, key); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// printPodErrors loops through pods, and prints pod errors if any.
+func (c Client) printPodErrors(ctx context.Context, depSelectors *metav1.LabelSelector, key types.NamespacedName) error {
+	// loop through pods and return specific error message.
+	podErrors := make(map[string]string)
+	podList := &corev1.PodList{}
+	podLabelSelectors, err := metav1.LabelSelectorAsSelector(depSelectors)
+	if err != nil {
+		return err
+	}
+	options := client.ListOptions{
+		LabelSelector: podLabelSelectors,
+		Namespace:     key.Namespace,
+	}
+	if err := c.KubeClient.List(ctx, podList, &options); err != nil {
+		return fmt.Errorf("error getting Pods: %v", err)
+	}
+	for _, p := range podList.Items {
+		if p.Status.Phase != corev1.PodSucceeded {
```

Comment:
`Succeeded` means the Pod finished and exited. For the operator pods, they should be running all the time to monitor the CRs, so, the expected status should be `Running`, not the `Succeeded`.